java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -813,7 +813,8 @@ public FlightClient build() {
 
       builder
           .maxTraceEvents(MAX_CHANNEL_TRACE_EVENTS)
-          .maxInboundMessageSize(maxInboundMessageSize);
+          .maxInboundMessageSize(maxInboundMessageSize)
+          .maxInboundMetadataSize(maxInboundMessageSize);
```

Comment:
I've noticed this is required for some implementations that try to return a relatively large error message. Without this, clients like DBeaver throw odd errors instead of giving the actual error message.